### PR TITLE
Step backwards version of hex2num

### PR DESCRIPTION
**Pull request by ninjanomnom**
I got interested so I just quickly threw together a better hex2num.

I tried a static list version and just a modified current version.

https://pastebin.com/dbPJ00j3

| Proc Name | Self CPU | Total CPU | Real Time | Calls |
| --- | --- | --- | --- | --- |
| /proc/goon_hex2num | 4.016 | 4.035 | 4.134 | 1000000 |
| /proc/my_list_hex2num | 3.640 | 3.655 | 3.761 | 1000000 |
| /proc/old_hex2num | 3.639 | 3.665 | 3.771 | 1000000 |
| /proc/my_modified_hex2num | 2.952 | 2.967 | 3.043 | 1000000 |

This also tested to ensure the values outputted are the same as before and makes it runtime when invalid characters are found

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -14,30 +14,25 @@
...
-						break
-					else
-						continue
-			. *= 16
-			. += num
-		if(negative)
-			. *= -1
-	return .
+	if(!istext(hex))
+		return
```
Comment (ninjanomnom) [inline code/__HELPERS/type2type.dm]: I feel like it'd be better to let it just have errors instead of this in here, is there anywhere that needs this?

Comment (Jordie0608) [reply code/__HELPERS/type2type.dm]: Sounds good

File: code/__HELPERS/type2type.dm [hunk 2]
```diff
@@ -14,30 +14,22 @@
...
+			if(97 to 102)
+				num -= 87	//a-f
+			if(65 to 70)
+				num -= 55	//A-F
+			if(45)
+				return . * -1 // -
+			else
+				CRASH("Malformed hex number")
+
+		. += num * (16 ** place++)
```
Comment (SpaceManiac) [inline code/__HELPERS/type2type.dm]: could you start with `place = 1` and `place *= 16` here?

**Comment by ninjanomnom** [comment]
We got the goon version fixed which means it can be fairly compared in profiling. ~~Surprisingly~~ it's slower than even the proc it was supposed to replace.

File: code/__HELPERS/type2type.dm (hunk 2 above)
Comment (ninjanomnom) [reply code/__HELPERS/type2type.dm]: Nice catch

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**